test/client-test.js

```diff
@@ -1834,6 +1834,93 @@ describe('Client posting complex body', () => {
       });
     }, baseUrl);
   });
+
+
+  it('should resolve promise on async invoke postProcess', function (done) {
+    soap.createClient(__dirname + '/wsdl/complex/registration-common.wsdl', function (err, client) {
+      if (err) {
+        return void done(err);
+      }
+      assert.ok(client);
+
+      var requestBody = {
+        id: 'ID00000000000000000000000000000000',
+        lastName: 'Doe',
+        firstName: 'John',
+        dateOfBirth: '1970-01-01',
+        correspondenceLanguage: 'ENG',
+        emailAddress: 'jdoe@doe.com',
+        lookupPermission: 'ALLOWED',
+        companyAddress: {
+          address: {
+            streetName: 'Street',
+            postalCode: 'Code',
+            city: 'City',
+            countryCode: 'US'
+          },
+          companyName: 'ACME'
+        }
+      }
+
+      client.registerUser(requestBody, {
+        postProcess: async (xml) => {
+          const testString = await new Promise(resolve => setTimeout(resolve("Jane"), 3000));
```

Comment:
This resolves immediately, so does not really simulate an async operation. 
I think it should be something like this 
```suggestion
          const testString = await new Promise(resolve => setTimeout(() => resolve("Jane"), 50));
```

What do you think?